google-cloud-spanner/src/main/java/com/google/cloud/spanner/SessionPoolAsyncTransactionManager.java

```diff
@@ -239,4 +244,22 @@ public TransactionState getState() {
       return txnState;
     }
   }
+
+  public ApiFuture<CommitResponse> getCommitResponse() {
+    synchronized (lock) {
+      Preconditions.checkState(
+          txnState == TransactionState.COMMITTED,
+          "commit can only be invoked if the transaction was successfully committed");
```

Comment:
Should this be "getCommitResponse can only be invoked..."?

Reply:
No, this is in the actual commit method. A client application could for example call `commit()` after calling `rollback` or before calling `begin`. This error will be thrown in those cases.